### PR TITLE
Fix leaking goroutines in multiple integration tests

### PR DESCRIPTION
**Pull request by wojtek-t**
Ref https://github.com/kubernetes/kubernetes/issues/108483

Note to reviewer: I recommend reviewing commits one by one - they are all independent.

```release-note
NONE
```

/kind cleanup
/priority important-longterm
/sig api-machinery

### CODE REVIEW

File: test/integration/apiserver/openapi/openapiv3_test.go [hunk 1]
```diff
@@ -46,11 +46,10 @@ import (
 
 func TestOpenAPIV3SpecRoundTrip(t *testing.T) {
 	defer featuregatetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, genericfeatures.OpenAPIV3, true)()
-	controlPlaneConfig := framework.NewIntegrationTestControlPlaneConfigWithOptions(&framework.ControlPlaneConfigOptions{})
-	controlPlaneConfig.GenericConfig.OpenAPIConfig = framework.DefaultOpenAPIConfig()
-	controlPlaneConfig.GenericConfig.OpenAPIV3Config = framework.DefaultOpenAPIV3Config()
```
Comment (aojea) [inline test/integration/apiserver/openapi/openapiv3_test.go:51]: don't you need to pass this to the StartTestServer() function 
```
				ModifyServerConfig: func(config *controlplane.Config) {
					config.GenericConfig.OpenAPIConfig = framework.DefaultOpenAPIConfig()
```

File: test/integration/apiserver/flowcontrol/concurrency_test.go [hunk 2]
```diff
@@ -150,16 +142,14 @@ func TestPriorityLevelIsolation(t *testing.T) {
 	if (noxu1RequestCount/2) > noxu2RequestCount || (noxu2RequestCount/2) > noxu1RequestCount {
 		t.Errorf("imbalanced requests made by noxu1/2: (%d:%d)", noxu1RequestCount, noxu2RequestCount)
 	}
+
+	t.Errorf("DUMP")
```
Comment (aojea) [inline test/integration/apiserver/flowcontrol/concurrency_test.go:146]: :upside_down_face: testing ?

File: test/integration/apiserver/openapi/openapiv3_test.go (hunk 1 above)
Comment (wojtek-t) [reply test/integration/apiserver/openapi/openapiv3_test.go:51]: You don't - that's a default...

**Comment by wojtek-t** [comment]
@aojea - PTAL

**Pull request merged**